db/migrate/20221003205400_create_coupons.rb

```diff
@@ -0,0 +1,11 @@
+class CreateCoupons < ActiveRecord::Migration[6.1]
+  def change
+    create_table :coupons do |t|
+      t.string :code, null: false
+      t.float :discount, null: false
+      t.belongs_to :order, foreign_key: true, index: false
```

Comment:
try index: {.unique: true }

I don't actually remember if it works, just try. If it does - ok, If does not - keep your code

Reply:
fixed